dashboard/app/models/user.rb

```diff
@@ -537,6 +537,7 @@ def sections
   validates_uniqueness_of :username, allow_blank: true, case_sensitive: false, on: :create, if: -> {errors.blank?}
   validates_uniqueness_of :username, case_sensitive: false, on: :update, if: -> {errors.blank? && username_changed?}
   validates_presence_of :username, if: :username_required?
+  validate :check_profanity
```

Comment:
We probably need an `if username_changed?` check here. Otherwise, this validation will run on any update, potentially causing an unintentional failure because an existing user already has a profane username.